### PR TITLE
Smart reader interface added

### PR DESCRIPTION
**Pull request by wkucjas**

### CODE REVIEW

**Comment by wkucjas** [comment]
How to avoid adding these two merge pull requests?

File: src/agent/SmartReader.h [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#pragma once
+#include <memory>
+#include <string>
+
+class SmartReader
+{
+public:
+
+	SmartReader();
+	std::unique_ptr<SmartReader> GetProperSMARTReader();
```
Comment (Kicer86) [inline src/agent/SmartReader.h]: I believe, this should be a method in a SMARTReadersFactory or something similar.

File: src/agent/SmartReader.h [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#pragma once
+#include <memory>
+#include <string>
+
+class SmartReader
```
Comment (Kicer86) [inline src/agent/SmartReader.h]: This should be an interface being returned from factory

File: src/agent/SmartReader.h [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+{
+public:
+
+	SmartReader();
+	std::unique_ptr<SmartReader> GetProperSMARTReader();
+	virtual std::string ReadSMARTData();
```
Comment (Kicer86) [inline src/agent/SmartReader.h]: pure virtual

**Review by Kicer86 — CHANGES_REQUESTED** [review]
Prepare factory and interface form SMART Reader

**Comment by wkucjas** [comment]
Changes added.

File: src/agent/LinuxSmartReader.h [hunk 4]
```diff
@@ -0,0 +1,11 @@
+#pragma once
```
Comment (Kicer86) [inline src/agent/LinuxSmartReader.h:1]: double pragma

File: src/agent/SmartReader.h [hunk 5]
```diff
@@ -0,0 +1,11 @@
+#pragma once
+#include <memory>
+#include <string>
+
+class SmartReader
```
Comment (Kicer86) [inline src/agent/SmartReader.h]: Interfaces usually start with "I" in names

File: src/agent/WindowsSmartReader.h [hunk 6]
```diff
@@ -0,0 +1,9 @@
+#pragma once
+#include "SmartReader.h"
+
+class WindowsSmartReader : public SmartReader
+{
+
+	std::string ReadSMARTData();
```
Comment (Kicer86) [inline src/agent/WindowsSmartReader.h]: mark with override

File: src/agent/LinuxSmartReader.h [hunk 7]
```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+#pragma once
+#include "SmartReader.h"
+
+class LinuxSmartReader : public SmartReader
+{
+
+	std::string ReadSMARTData();
```
Comment (Kicer86) [inline src/agent/LinuxSmartReader.h]: mark with override

**Review by Kicer86 — CHANGES_REQUESTED** [review]
some minor improvements required

**Comment by wkucjas** [comment]
Done.

**Review by Kicer86 — APPROVED** [review]

**Pull request merged**